content/doc/book/pipeline/syntax.adoc

```diff
@@ -0,0 +1,851 @@
+be defined as environment variables for the all steps, or stage-specific steps,
+depending on where the `environment` directive is located within the Pipeline.
+
+This directive supports a special helper method `credentials()` which can be
+used to access pre-defined Credentials by their identifier in the Jenkins
+environment. For Credentials which are of type "Secret Text", the
+`credentials()` method will ensure that the environment variable specified
+contains the Secret Text contents. For Credentials which are of type "Standard
+username and password", the environment variable specified will be set to
+`username:password` and two additional environment variables will be
+automatically be defined: `MYVARNAME_USR` and `MYVARNAME_PSW` respective.
+
+
+[cols="^10h,>90a",role=syntax]
+|===
+| Required
+| No
+
+| Parameters
+| _None_
+
+| Allowed
+| Inside the `pipeline` block, or within `stage` directives.
+|===
+
+[[environment-example]]
+===== Example
+
+[pipeline]
+----
+// Declarative //
+pipeline {
+    agent any
+    environment { /* <1> */
+        CC = 'clang'
+    }
+    stages {
+        stage('Example') {
+            environment { /* <2> */
+                AN_ACCESS_KEY = credentials('my-prefined-secret-text') /* <3> */
+            }
+            steps {
+                sh 'printenv'
+            }
+        }
+    }
+}
+----
+<1> An `environment` directive used in the top-level `pipeline` block will
+apply to all steps within the Pipeline.
+<2> An `environment` directive defined within a `stage` will only apply the
+given environment variables to steps within the `stage`.
+<3> The `environment` block has a helper method `credentials()` defined which
+can be used to access pre-defined Credentials by their identifier in the
+Jenkins environment.
+
+==== options
+
+The `options` directive allows configuring Pipeline-specific options from
+within the Pipeline itself. Pipeline provides a number of these options, such
+as `buildDiscarder`, but they may also be provided by plugins, such as
+`timestamps`.
+
+
+[cols="^10h,>90a",role=syntax]
+|===
+| Required
+| No
+
+| Parameters
+| _None_
+
+| Allowed
+| Only once, inside the `pipeline` block.
+|===
+
+===== Available Options
+
+buildDiscarder:: Persist artifacts and console output for the specific number
+of recent Pipeline runs. For example: `options { buildDiscarder(logRotator(numToKeepStr: '1')) }`
+disableConcurrentBuilds:: Disallow concurrent executions of the Pipeline. Can
+be useful for preventing simultaneous accesses to shared resources, etc. For
+example: `options { disableConcurrentBuilds() }`
+skipDefaultCheckout:: Skip checking out code from source control by default in
+the `agent` directive. For example: `options { skipDefaultCheckout() }`
+timeout:: Set a timeout period for the Pipeline run, after which Jenkins should
+abort the Pipeline. For example: `options { timeout(time: 1, unit: 'HOUR') }`
```

Comment:
At some point, it would be nice to have a full reference here — i.e. can I use 'MILLISECONDS', 'YEARS' etc.?

Or we could mention that the full documentation can be found on the Steps Reference page, and link there from here. 
(Though just now I see that the `timeout` step, and `JobProperty`s like `logRotator` aren't listed on that reference page)